File: apps/backend/internal/domain/types/org.go
package types

type Platform string

const (
	PlatformFeishu   Platform = "feishu"
	PlatformDingtalk Platform = "dingtalk"
	PlatformWecom    Platform = "wecom"
)

type DataSourceStatus struct {
	Platform         *Platform     `json:"platform"`
	Connected        bool          `json:"connected"`
	LastImport       *string       `json:"lastImport"`
	LastImportResult *ImportResult `json:"lastImportResult"`
}

type ImportResult struct {
	SuccessMembers     int             `json:"successMembers"`
	SuccessDepartments int             `json:"successDepartments"`
	Failures           []ImportFailure `json:"failures"`
}

type ImportFailure struct {
	ID         string `json:"id"`
	Name       string `json:"name"`
	EmployeeID string `json:"employeeId"`
	Reason     string `json:"reason"`
}

type SyncConfig struct {
	Enabled                   bool   `json:"enabled"`
	StartTime                 string `json:"startTime"`
	FrequencyHours            int    `json:"frequencyHours"`
	DeleteMemberThreshold     int    `json:"deleteMemberThreshold"`
	DeleteDepartmentThreshold int    `json:"deleteDepartmentThreshold"`
	NotifyPhone               bool   `json:"notifyPhone"`
	NotifyEmail               bool   `json:"notifyEmail"`
	NotifyIm                  bool   `json:"notifyIm"`
}

type SyncLog struct {
	ID     string `json:"id"`
	Time   string `json:"time"`
	Type   string `json:"type"`
	Result string `json:"result"`
	Detail string `json:"detail"`
}

type Department struct {
	ID          string       `json:"id"`
	Name        string       `json:"name"`
	ParentID    *string      `json:"parentId"`
	Children    []Department `json:"children,omitempty"`
	MemberCount int          `json:"memberCount"`
}

type BatchImportRow struct {
	Name           string `json:"name"`
	Phone          string `json:"phone"`
	Email          string `json:"email"`
	DepartmentName string `json:"departmentName"`
}

type MemberBatchImportResult struct {
	Imported int                        `json:"imported"`
	Failures []MemberBatchImportFailure `json:"failures"`
}

type MemberBatchImportFailure struct {
	Row    int    `json:"row"`
	Reason string `json:"reason"`
}

type Permission struct {
	ID    string `json:"id"`
	Name  string `json:"name"`
	Group string `json:"group"`
}

type DataSourceSearchResult struct {
	Name       string `json:"name"`
	Department string `json:"department"`
	MappingOK  bool   `json:"mappingOk"`
}

type DataSourceTestResult struct {
	Success bool    `json:"success"`
	Message *string `json:"message,omitempty"`
}

type BatchInviteResult struct {
	Sent int `json:"sent"`
}

File: apps/frontend/src/api/types/org.ts
export type Platform = 'feishu' | 'dingtalk' | 'wecom'

export interface FeishuCredential {
  platform: 'feishu'
  appId: string
  appSecret: string
}

export interface DingtalkCredential {
  platform: 'dingtalk'
  corpId: string
  appKey: string
  appSecret: string
}

export interface WecomCredential {
  platform: 'wecom'
  corpId: string
  secret: string
  agentId: string
}

export type Credential = FeishuCredential | DingtalkCredential | WecomCredential

export interface DataSourceStatus {
  platform: Platform | null
  connected: boolean
  lastImport: string | null
  lastImportResult: ImportResult | null
}

export interface ImportResult {
  successMembers: number
  successDepartments: number
  failures: ImportFailure[]
}

export interface ImportFailure {
  id: string
  name: string
  employeeId: string
  reason: string
}

export interface SyncConfig {
  enabled: boolean
  startTime: string
  frequencyHours: 6 | 12 | 24
  deleteMemberThreshold: number
  deleteDepartmentThreshold: number
  notifyPhone: boolean
  notifyEmail: boolean
  notifyIm: boolean
}

export interface SyncLog {
  id: string
  time: string
  type: 'scheduled' | 'manual'
  result: 'success' | 'partial_failure' | 'failure'
  detail: string
}

export interface Department {
  id: string
  name: string
  parentId: string | null
  children?: Department[]
  memberCount: number
  externalId?: string
  source?: 'imported' | 'manual'
  managerId?: string
}

export type MemberStatus = 'active' | 'disabled' | 'pending'

export type MemberSource =
  | 'manual'
  | 'csv'
  | 'feishu'
  | 'dingtalk'
  | 'wecom'
  | 'invited'
  | 'imported'

export type RegistrationChannel = 'sms' | 'email' | 'admin_link'

export interface Member {
  id: string
  companyId: string
  alias: string
  avatar: string
  departmentId: string
  departmentName: string
  status: MemberStatus
  roles: string[]
  source: MemberSource
  registrationChannel?: RegistrationChannel
  externalId?: string
  employeeId?: string
  jobTitle?: string
  hireDate?: string
  // Read-only fields from users table (JOIN)
  name?: string
  phone?: string
  email?: string
}

// Request body for POST /org/members
export interface CreateMemberInput {
  user: { name: string; phone: string; email: string }
  member: {
    alias?: string
    departmentId: string
    employeeId?: string
    jobTitle?: string
    hireDate?: string
  }
}

// Request body for PUT /org/members/:id
export interface UpdateMemberInput {
  alias?: string
  departmentId?: string
  employeeId?: string
  jobTitle?: string
  hireDate?: string
  roles?: string[]
  status?: string
}

// Request body for PUT /org/members/:id/user
export interface UpdateMemberUserInput {
  name?: string
  phone?: string
  email?: string
}

export interface BatchImportRow {
  name: string
  phone: string
  email: string
  departmentName: string
}

export interface MemberBatchImportResult {
  imported: number
  failures: { row: number; reason: string }[]
}

export interface Role {
  id: string
  name: string
  type: 'preset' | 'custom'
  permissions: string[]
  memberCount: number
}

export interface Permission {
  id: string
  name: string
  group: string
}

export interface FieldMapping {
  sourceField: string
  sourceLabel: string
  targetField: string
  required: boolean
}

export interface FieldMappingConfig {
  platform: Platform
  mappings: FieldMapping[]
}

export interface MappingTestResult {
  success: boolean
  preview: Record<string, string>
  errors: string[]
}
